packages/framework/client-logger/external-telemetry/src/container/containerTelemetry.ts

```diff
@@ -0,0 +1,170 @@
+ */
+export interface ContainerConnectedTelemetry extends IContainerTelemetry {
+	eventName: "container.connected";
+}
+
+/**
+ * The container "disconnected" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer}
+ * becomes disconnected from the Fluid service.
+ *
+ * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+ *
+ * @see
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.connectionState}
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.disconnect}
+ *
+ * @beta
+ */
+export interface ContainerDisconnectedTelemetry extends IContainerTelemetry {
+	eventName: "container.disconnected";
+}
+
+/**
+ * The container "closed" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer}
+ * is closed, which permanently disables the container.
+ *
+ * @remarks Listener parameters:
+ *
+ * - `error`: If the container was closed due to error, this will contain details about the error that caused it.
+ *
+ * @see {@link @fluidframework/container-definitions#IContainer.close}
+ *
+ * @beta
+ */
+export interface ContainerClosedTelemetry extends IContainerTelemetry {
+	eventName: "container.closed";
+	error?: ICriticalContainerError;
+}
+
+/**
+ * The container "attaching" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer}'s
+ * {@link @fluidframework/container-definitions#AttachState.Attaching | attaching} process is complete and the container is {@link @fluidframework/container-definitions#AttachState.Attached | attached} to the Fluid service.
+ *
+ * @see
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.attachState}
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.attach}
+ *
+ * @beta
+ */
+export interface ContainerAttachingTelemetry extends IContainerTelemetry {
+	eventName: "container.attaching";
+}
+
+/**
+ * The container "attached" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when a {@link @fluidframework/container-definitions#AttachState.Detached | detached}
+ * container begins the process of {@link @fluidframework/container-definitions#AttachState.Attaching | attached} to the Fluid service.
+ *
+ * @see
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.attachState}
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.attach}
```

Comment:
TSDoc docs suggests this is how we should use `@see`, can we update the others?
```suggestion
 * @see More details about {@link @fluidframework/container-definitions#IContainer.attachState | the container's attach state}
 * @see {@link @fluidframework/container-definitions#IContainer.attach | the function to attach a container}
```

Reply:
Updated.